gcloud/datastore/helpers.py

```diff
@@ -45,12 +46,17 @@ def entity_from_protobuf(pb):
     :returns: The entity derived from the protobuf.
     """
     key = key_from_protobuf(pb.key)
-    entity = Entity(key=key)
+    entity_props = {}
+    exclude_from_indexes = []
 
     for property_pb in pb.property:
         value = _get_value_from_property_pb(property_pb)
-        entity[property_pb.name] = value
+        entity_props[property_pb.name] = value
+        if not property_pb.value.indexed:
```

Comment:
Ah good catch. It somewhat introduces a flaw in our abstraction since with our library the entire list must be indexed or un-indexed but if stored elsewhere, that restriction does not hold.

WDYT?

----

This is a place where the `gcloud-node` way of using a dictionary is simpler / more correct than the `Entity` class. For example:

```python
e = {}
e['foo'] = u'bar'
e['baz'] = {'value': 10, 'excludeFromIndexes': True}
e['qux'] = [42.2, {'value': '\x00', 'excludeFromIndexes': True}, 'other thing']
```